experimental/dds/tree2/src/feature-libraries/sequence-field/utils.ts

```diff
@@ -430,12 +443,12 @@ export class DetachedNodeTracker {
 							this.equivalences.push({
 								old: v,
 								new: {
-									rev:
+									revision:
 										change.rollbackOf ??
 										mark.revision ??
 										change.revision ??
 										fail("Unable to track detached nodes"),
-									index: k,
+									id: brand((mark.id as number) + (k - index)),
```

Comment:
Are all these casts from `ChangesetLocalId` to `number` really necesssary?

Reply:
The compiler is fine with it, but we have a [lint rule](https://typescript-eslint.io/rules/restrict-plus-operands/) which complains.

Reply:
I see. It's a little lame that the rule doesn't allow `Foo & number + number`, but it's better than no check at all. Resolving this thread.